### PR TITLE
refactor: 使用coreMqtt重构

### PR DESCRIPTION
**Pull request by Ryan-CW-Code**
审查代码

## Summary by CodeRabbit

* **New Features**
  * New core MQTT serializer, batch subscribe/unsubscribe and unified client APIs; standardized logging interface.

* **Improvements**
  * Better thread-safety, memory handling, keepalive and reconnect behavior; streamlined network/send/receive flows.

* **Bug Fixes**
  * Fixed race conditions (packet IDs, acks), duplicate connect/disconnect edge cases and related stability issues.

* **Refactor**
  * Replaced legacy MQTT stack with modular core serializer and reorganized client/utility/transport layers.

* **Tests**
  * Added extensive automated tests for publish/subscribe, reconnect, keep‑alive and multi‑thread scenarios.

* **Chores**
  * Added formatting and static-analysis configs; updated build and ignore settings.

### CODE REVIEW

File: mqttclient/RyanMqttUtile.c [hunk 1]
```diff
@@ -62,30 +69,52 @@ RyanMqttError_e RyanMqttSetPublishDup(char *headerBuf, uint8_t dup)
...
+	}
+
+	// 发送超时
+	if (offset != recvLen)
+	{
+		return RyanMqttRecvPacketTimeOutError;
+	}
+
+	return RyanMqttSuccessError;
 }
```
[the remark this answers is not in the document]
Comment (Ryan-CW-Code) [reply mqttclient/RyanMqttUtile.c:118]: 帮我生成多线程测试代码

File: mqttclient/RyanMqttThreadProcessPacket.c [hunk 2]
```diff
@@ -0,0 +1,591 @@
...
+			}
+			platformMutexUnLock(client->config.userData, &client->msgHandleLock);
+		});
+	}
+
+	uint32_t ackMsgIndex = 0;
+	RyanMqttQos_e subscriptionQos = RyanMqttSubFail;
+	const uint8_t *pStatusStart = &pIncomingPacket->pRemainingData[sizeof(uint16_t)];
+
+	// todo 这里效率非常低，订阅属于用的少的功能，暂时可以接受
```
[the remark this answers is not in the document]
Comment (Ryan-CW-Code) [reply mqttclient/RyanMqttThreadProcessPacket.c:316]: 帮我生成多线程测试代码

[the remark this answers is not in the document]
Comment (Ryan-CW-Code) [reply mqttclient/RyanMqttThreadProcessPacket.c:316]: 帮我写一个测试订阅主题通配符的测试代码

File: mqttclient/RyanMqttUtileAck.c [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+		if (RyanMqttSuccessError == result)
+		{
+			RyanMqttAckListRemoveToAckList(client, ackHandler);
+			RyanMqttAckHandlerDestroy(client, ackHandler);
+			continue;
+		}
+
+		break;
+	}
+}
```
[the remark this answers is not in the document]
Comment (Ryan-CW-Code) [reply mqttclient/RyanMqttUtileAck.c:279]: 这个只有在订阅和取消订阅函数和mqtt线程处理订阅和非订阅消息的时候才会执行，订阅的取消订阅的packetid都是不一致的，而且顺序一定是先发送订阅或者取消订阅消息，才会轮到mqtt线程接收订阅回复消息检查条目并调用

File: mqttclient/RyanMqttThread.c [hunk 4]
```diff
@@ -22,615 +23,191 @@ void RyanMqttRefreshKeepaliveTime(RyanMqttClient_t *client)
...
+				RyanMqttEventMachine(client, RyanMqttEventAckRepeatCountWarning, (void *)ackHandler);
+				continue;
+			}
+
+			// 重发数据事件回调
+			RyanMqttEventMachine(client, RyanMqttEventRepeatPublishPacket, (void *)ackHandler);
+
+			//? 发送失败也是重试,所以这里不进行错误判断
+			RyanMqttSendPacket(client, ackHandler->packet, ackHandler->packetLen); // 重新发送数据
+
```
[the remark this answers is not in the document]
Comment (Ryan-CW-Code) [reply mqttclient/RyanMqttThread.c:171]: 能进到这个case的全是pub消息

File: mqttclient/RyanMqttThread.c [hunk 5]
```diff
@@ -22,615 +26,200 @@ void RyanMqttRefreshKeepaliveTime(RyanMqttClient_t *client)
...
+		{
+			return RyanMqttSuccessError;
+		}
+
+		// 节流时间内不发送心跳报文
+		if (RyanMqttTimerRemain(&client->keepaliveThrottleTimer))
+		{
+			return RyanMqttSuccessError;
+		}
+	}
```
[the remark this answers is not in the document]
Comment (Ryan-CW-Code) [reply mqttclient/RyanMqttThread.c:63]: 要的效果就是这样的把

**Comment by Ryan-CW-Code** [comment]
@coderabbitai 帮我生成详细的摘要和修改记录

**Comment by Ryan-CW-Code** [comment]
@coderabbitai 帮我生成详细的摘要和修改记录

**Comment by Ryan-CW-Code** [comment]
@coderabbitai  再重点介绍下代码质量提升（使用了clang-tidy和cppcheck）和安全性和可靠性增强
